src/compile/mark/mark.ts

```diff
@@ -134,33 +137,16 @@ function parseNonPathMark(model: UnitModel) {
         model.has(ORDER) ?
           // if non-stacked, detail field determines the layer order of each mark
           { transform: [{type:'sort', by: sortBy(model)}] } :
+          {},
+        mark === LABEL && referenceMark ?
```

Comment:
If you specify order,  you should be able to order the label, provided that your label transform might still allow some occlusion?  

(Order should come after label - I think.)